sources/packages/backend/apps/api/src/route-controllers/student-scholastic-standings/student-scholastic-standings.institutions.controller.ts

```diff
@@ -139,4 +160,84 @@ export class ScholasticStandingInstitutionsController extends BaseController {
+   * @param validationOnly if true, will execute all validations and return the
+   * errors and warnings. These validations are the same executed during the
+   * final creation process. If not present or false, the file will be processed
+   * and the records will be inserted.
+   * @returns when successfully executed, the list of all withdrawn application ids.
+   * When an error happen it will return all the records (with the error) and
+   * also a user friendly description of the errors to be fixed.
+   */
+  @ApiBadRequestResponse({
+    description:
+      "Error while parsing text file or " +
+      "one or more text data fields received are not in the correct format or " +
+      "there are no records to be imported.",
+  })
+  @ApiUnprocessableEntityResponse({
+    description:
+      "Application withdrawal has invalid data or " +
+      "some error happen with one or more application withdrawal and the entire process was aborted.",
+  })
+  @IsInstitutionAdmin()
+  @Post("application-bulk-withdrawal")
+  @UseInterceptors(
+    FileInterceptor("file", {
+      limits: uploadLimits(
+        MAX_UPLOAD_FILES,
+        APPLICATION_BULK_WITHDRAWAL_MAX_UPLOAD_PARTS,
+        APPLICATION_BULK_WITHDRAWAL_UPLOAD_MAX_FILE_SIZE,
+      ),
+      fileFilter: textFileFilter,
+    }),
+  )
+  async bulkWithdrawal(
+    @UserToken() userToken: IInstitutionUserToken,
+    @UploadedFile() file: Express.Multer.File,
+    @Query("validation-only", new DefaultValuePipe(false), ParseBoolPipe)
+    validationOnly: boolean,
+  ): Promise<PrimaryIdentifierAPIOutDTO[]> {
+    // Read the entire file content.
+    const fileContent = file.buffer.toString();
+    // Convert the file raw content into text models.
+    let textModels: ApplicationWithdrawalTextModel[];
+    try {
+      textModels =
+        this.applicationWithdrawalImportTextService.readText(fileContent);
+    } catch (error: unknown) {
+      let errorMessage = "Error while parsing text file.";
+      if (
+        error instanceof CustomNamedError &&
+        error.name === APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR
+      ) {
+        errorMessage = error.message;
+      }
+      throw new BadRequestException(
+        new ApiProcessError(
+          errorMessage,
+          APPLICATION_WITHDRAWAL_INVALID_TEXT_FILE_ERROR,
+        ),
+      );
+    }
+    // Validate the text models.
+    const textValidations =
+      this.applicationWithdrawalImportTextService.validateTextModels(
+        textModels,
+      );
+    // Assert successful validation.
+    this.scholasticStandingControllerService.assertTextValidationsAreValid(
+      textValidations,
+    );
+
+    if (validationOnly) {
+      // If the endpoint is called only to perform the validation and no error was found
+      // return an empty array because no record will be created.
+      return [];
+    }
+
+    // TODO create a block to do database updates for Application withdrawal.
```

Comment:
Are you going to work on this in another PR?

Reply:
No as part of the next story #2112